### PR TITLE
Make the code actually work as well as more flexible

### LINKED ISSUE #16: Use YAML inventory and group variables files to simplify use of this project
# 💡 Summary #

I think the use of this project can be enhanced by using a YAML inventory file and group variables stored in the project's root directory, and I think it makes sense to modify the `README.md` instructions to reflect this.

## Motivation and context ##

When I recently used this project to create the credentialed users for our compliance scans, I created an `ansible-manage-user/inventory.yml` file that looked like this:
```yaml
all:
    children:
        openvpn:
            hosts:
                i-0123456789abcdef0:
        freeipa:
            hosts:
                i-0123456789abcdef1:
                i-0123456789abcdef2:
                i-0123456789abcdef3:
```

Note that this more easily allows one to create or destroy the user for a particular group of servers.

I also created an `ansible-manage-user/group_vars` directory, which contain a single file named `all.yml` that looks like:
```yaml
allow_sudo: true
ansible_become: yes
username: abcdef
ssh_public_key: ssh-rsa blahblahblah abcdef
```

With these two files in place, I can run the "create" playbook like this:
```bash
$ AWS_PROFILE=cool-sharedservices-startstopssmsession AWS_SHARED_CREDENTIALS_FILE=~/.aws/production_credentials AWS_DEFAULT_REGION=us-east-1 ansible-playbook --inventory=inventory.yml create/playbook.yml
```
And I can run the "delete" playbook (whose name perturbs me since "destroy" is the opposite of "create") like this:
```bash
$ AWS_PROFILE=cool-sharedservices-startstopssmsession AWS_SHARED_CREDENTIALS_FILE=~/.aws/production_credentials AWS_DEFAULT_REGION=us-east-1 ansible-playbook --inventory=inventory.yml delete/playbook.yml
```

To my mind, these changes make this project quite a bit easier to use.

## Acceptance criteria ##

- [x] The instructions in the `README.md` file are edited to reflect the above changes.

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request makes several improvements to the existing code:
* Require that the `ssh` public key be specified as an input parameter.  This is simpler and more flexible than including the `lookup('aws_ssm', ...)` jazz in the actual playbook.  The user can still specify the input parameter as the result of a `lookup(...)` if desired; this is precisely what is done [here](https://github.com/cisagov/teamserver-packer/blob/c1b5083828feb8255c40f3b2e5a551f7646d41b8/src/vnc.yml#L6-L13), for example
* Update `README.md` to address the change explained in the previous bullet.
* Update `README.md` to use the preferred YAML format for the Ansible inventory file.
* Update `README.md` to specify the Ansible input variables using `group_vars` instead of the command line.

## 💭 Motivation and context ##

The first and second bullets above resolve #16.  The other bullets are usability improvements.

## 🧪 Testing ##

I used these changes to successfully create the COOL SharedServices compliance scan users.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,47 +20,61 @@ that you plan to manage!
...
+* Ansible supports more complicated inventory management.  If you have
+  a need for that, consult the [Ansible
   documentation](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html)
   for more information.
-* When adding a new user, the user's public ssh key must be stored as a
-  parameter in
-  [AWS SSM](https://docs.aws.amazon.com/systems-manager/latest/userguide/what-is-systems-manager.html).
+* You must create a directory `group_vars` at the root of this project
+  that contains files with the Ansible variable values for each group
+  specified in the inventory; for example, so specify the same values
```
Comment (dav3r) [inline README.md]: Typo:
```suggestion
  specified in the inventory; for example, to specify the same values
```

**Review by dav3r — APPROVED** [review]
This is much nicer than how it was set up previously!  Thanks for cleaning up my hacked-together implementation.  🚀

File: README.md [hunk 2]
```diff
@@ -20,47 +20,61 @@ that you plan to manage!
...
+    children:
+      group1:
+        hosts:
+          my_server:
+          my_database:
+      group2:
+        hosts:
+          sample.mydomain.com:
+          db.mydomain.com:
+      group2:
```
Comment (mcdonnnj) [inline README.md]: I don't think this is supposed to be a duplicate key.

```suggestion
      group3:
```

**Review by mcdonnnj — APPROVED** [review]
Much more flexible! Thanks for straightening this out. I have one suggestion for your consideration.

File: README.md (hunk 2 above)
Comment (jsf9k) [reply README.md]: You are correct sir!

**Pull request merged**